controllers/user_controller.go

```diff
@@ -0,0 +1,75 @@
+package controllers
+
+import (
+	"context"
+
+	controlv1 "github.com/appuio/control-api/apis/v1"
+	"github.com/vshn/appuio-keycloak-adapter/keycloak"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// UserReconciler reconciles a User object
+type UserReconciler struct {
+	client.Client
+	Recorder record.EventRecorder
+	Scheme   *runtime.Scheme
+
+	Keycloak KeycloakClient
+}
+
+//+kubebuilder:rbac:groups=appuio.io,resources=users,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=appuio.io,resources=users/status,verbs=get;update;patch
+
+// Reconcile reacts on changes of users and mirrors these changes to Keycloak
+func (r *UserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+	log.V(4).WithValues("request", req).Info("Reconciling")
+
+	log.V(4).Info("Getting the User..")
+	user := controlv1.User{}
```

Comment:
The user controller is very easy, so maybe we actually don't need a test for it. But I generally prefer to at least have a simple smoke test to notice if we obviously broke something and as a base test for when the code does get more complicated.

Reply:
Totally agreed. Got lost in git.